src/main/java/Oenskeskyen/Controller/SessionController.java

```diff
@@ -0,0 +1,150 @@
+        if(user != null){
+            int userID = user.getId();
+            userService.saveWishList(wishListObj, userID);
+            return "redirect:/customer-page";
+        } else {
+            return "redirect:/login";
+        }
+    }
+
+    @GetMapping("/addWish/{wishListID}")
+    public String createNewWish(@PathVariable("wishListID") int wishListID, Model model){
+        Wish obj = new Wish();
+        model.addAttribute("obj", obj);
+        model.addAttribute("wishListID", wishListID);
+        return "addWish";
+    }
+
+    @PostMapping("/saveWish")
+    public String saveWish(@ModelAttribute Wish wishObj, @RequestParam("wishListID") int wishListID){
+        userService.saveWish(wishObj, wishListID);
+        return "redirect:/viewWishList/" + wishListID;
+    }
+
+    @PostMapping("/deleteWishList")
+    public String deleteWishList(int wishlistID){
+        userService.deleteWishList(wishlistID);
+        return "redirect:/customer-page";
+    }
+
+    @GetMapping("/login")
+    public String login(Model model){
+        return "login";
+    }
+
+    @PostMapping("/loginValidation")
+    public String loginCheck (@RequestParam("mail") String mail, @RequestParam("password") String password,
+                              HttpSession session, Model model){
+       User user = userService.getUser(mail);
+       if (user != null && user.getPassWord().equals(password)) {
+           session.setAttribute("user", user);
+           return "redirect:/customer-page";
+       } else {
+           model.addAttribute("error", "Invalid username or password");
+           return "login";
+       }
+    }
+
+    @GetMapping("/logout")
+    public String logout(HttpSession session) {
+        session.invalidate();
+        return "redirect:/login";
+    }
+
+    @GetMapping("/customer-page")
+    public String costumerPage(Model model, HttpSession session){
+        User user = (User) session.getAttribute("user");
+        if(user != null){
+            model.addAttribute("user", user);
+            model.addAttribute("getAllWishLists", userService.getAllWishLists(user.getId()));
+            return "customer-page";
+        } else {
+            return "redirect:/login";
+        }
+    }
+
+    @GetMapping("/viewWishList/{wishListID}")
+    public String viewWishList(@PathVariable("wishListID") int wishListID, Model model){
+        WishList wishList = userService.getWishListById(wishListID);
+        if(wishList != null){
+            model.addAttribute("wishlist", wishList);
+            model.addAttribute("getAllWishes", userService.getAllWishes(wishListID));
+            return "viewWishList";
+        } else {
+            return "redirect:/customer-page";
+        }
+    }
+
+    @GetMapping("/set_session_id")
+    public String setSessionId(HttpServletRequest request){
+        HttpSession session = request.getSession();
+        session.setAttribute("sessionId", session.getId());
+        return "redirect:";
+    }
+
```

Comment:
All logic that isnt related to the session and checking if the user is logged in should be in OenskeSkyenController. I will take care of it